src/Sarif.Viewer.VisualStudio.UnitTests/CodeFlowToTreeConverterTests.cs

```diff
@@ -173,5 +173,193 @@ public void CanConvertCodeFlowToTreeOnlyDeclarations()
             topLevelNodes[1].Location.Kind.Should().Be(AnnotatedCodeLocationKind.Declaration);
             topLevelNodes[2].Location.Kind.Should().Be(AnnotatedCodeLocationKind.Declaration);
         }
+
+        [Fact]
+        public void CheckBalance1()
+        {
+            var exceptionThrown = false;
+
+            var codeFlow = new CodeFlow
+            {
+                Locations = new List<AnnotatedCodeLocation>
+                {
+                    new AnnotatedCodeLocation
+                    {
+                        Kind = AnnotatedCodeLocationKind.Call
+                    }
+                }
+            };
+
+            try
+            {
+                List<CallTreeNode> topLevelNodes = CodeFlowToTreeConverter.Convert(codeFlow);
+            }
+            catch (System.ArgumentException)
+            {
+                exceptionThrown = true;
+            }
+
+            if (!exceptionThrown)
+            {
+                throw new System.Exception();
+            }
+
```

Comment:
FluentAssertions has a [lovely facility](https://github.com/dennisdoomen/fluentassertions/wiki#exceptions) for this:

```C#
Action action = () => CodeFlowToTreeConverter.Convert(codeFlow); // C# lambda syntax

action.ShouldThrow<ArgumentException>();
```